### DISCUSSION TITLE
Translations and future features

### DISCUSSION THREAD

**Discussion by MaderatorCreator** [diasurgical/DevilutionX#7039] [General]
### Feature Type

Quality of Life

### Describe

1. Translation of the final video
The final video always sounds in English, even if ru.mpq and po.mpq are installed. DevX also supports a huge number of languages ​​without voice acting. I propose to include in existing audio localizations the translation of videos that is in the original translations for vanilla. But it seems to me that this will not be easy and will not make much sense, because there are only two audio localizations. I suggest adding subtitles for the final video, which will change depending on the language settings of the application.

2. Translucent walls
In the Labyrinth on levels 1-8 and in the Crypt, all (or almost all) walls are translucent. This helps highlight the barrels, chests and loot lying right against the wall at the bottom of the screen. However, there are no translucent walls in levels 9-16 and in the Nest. This makes it a little difficult to collect loot.

3. New settings
It would be nice to be able to change the controls (already mentioned a bunch of times), run Diablo 2 LoD (in my dreams) and manage innovations in DevX. What if someone wanted to play the most ordinary Diablo? For example, without storage. There you can also put the ability to turn on/off bugs for vanilla Diablo/Hellfire. Vanilla Mod, heh.

4. PS1 version
Why not add the ability to play the PlayStation 1 version of Diablo? Sometimes I want to play this particular version, but I don’t want to bother with PS1 emulators. But I know it will be difficult - implementing new controls and adapting it to all platforms is crazy. Plus write code. Well, what if you do this someday?

5. Translation of settings
Some settings are only available in English. There are no lines in the .po files to translate these words, and I personally don’t understand how to add them.

**Comment by AJenbo** [comment]
Vanilla is only available in English and was never officially translated to any other languages. These are fan made.

**Comment by julealgon** [comment]
@Sergei3000yt , please avoid having a bunch of different unrelated things in the same issue. You are only making tracking more difficult by doing this.

**Reply by AJenbo** [reply]
I converted it to a discussion since it has multiple things and I can't even tell if they are suggestions or questions

**Comment by AJenbo** [comment]
12-16 have translucent wall. Caves/nest doesn't because of how the tiles are layd out

**Comment by AJenbo** [comment]
Diablo 1 will not run Diablo 2 LoD, different game, diffentent engine different mechanics, different formats. You might as well ask for it to run starcraft and tomb raider.

**Comment by AJenbo** [comment]
We do intent to split out some of the qol in to script mods which would allow you to better customize what changes to have in the game.

**Comment by AJenbo** [comment]
Regarding the PS1 version it isn't fully clear what aspect you are asking for. But we already support playing via gamepad and you can change game speed.

Adding colored lights and hot seat multiplayer would be nice, but will take time. The only missing part then is unlocked movement and hell horror monsters which could be a simple mod. Or maybe you are thinking of something else? We probably won't add support for the data format. Someone could make a ui mod if that is what you want.

**Comment by AJenbo** [comment]
> Translation of settings

Did you update the translation file from source? If it's still missing then report the specific text that isn't translatable

**Reply by AJenbo** [reply]
![image](https://github.com/diasurgical/devilutionX/assets/204594/b045d1e5-a716-4f6e-81f5-c958faed6452)

**Comment by AJenbo** [comment]
We do want to add support for subtitles for the few things that currently do not have text (there are a few spoken lines in game as well). But this hasn't been done yet.

**Comment by MaderatorCreator** [comment]
> Did you update the translation file from source?

Throws an error "Source code not available."
The following settings and their descriptions are not available in other languages:
• Sound → Resamplex
• Sound → Device
• Gameplay → Full quests in Multiplayer
• Gameplay → Show Item Graphics in Stores
• Gameplay → Show Item Labels
• Gameplay → Enable floating numbers
• Gameplay → Auto Oil Pickup
• Keymapping → Use health/mana potion
• Padmapping
• Padmapping → Primary action
• Padmapping → Secondary action
• Padmapping → Spell action
• Padmapping → Cancel action
• Padmapping → Move up/down/left/right
• Padmapping → Stand ground
• Padmapping → Toggle stand ground
• Padmapping → Use health/mana potion
• Padmapping → Move mouse up/down/left/right
• Padmapping → Left/Right mouse click
• Padmapping → Gamepad hotspell menu
• Padmapping → Gamepad menu navigator
• Padmapping → Toggle game menu



> Regarding the PS1 version it isn't fully clear what aspect you are asking for.

I meant the possibility of running the PS1 version on DevX and support for this version, say, of a keyboard and on-screen controls. The launch of the same Diablo for PS1 - fuzzy, with broken seeds, crooked animations, bad sound, reflection from water, and so on.

**Reply by AJenbo** [reply]
> I meant the possibility of running the PS1 version on DevX and support for this version, say, of a keyboard and on-screen controls. The launch of the same Diablo for PS1 - fuzzy, with broken seeds, crooked animations, bad sound, reflection from water, and so on.

I'm not sure I follow. Are you saying we should make DevilutionX run on a PS1? The original Diablo had to be cut down a lot to run on PS1, this isn't viable for us to do.

> with broken seeds

?

> crooked animations, bad sound

This does not sound like something we want to support.

> reflection from water

This could be interesting to add, same with birds

**Reply by AJenbo** [reply]
> Throws an error "Source code not available."

Maybe you didn't download the source? Which languages are you looking at and for which version? There might be a 1.5.3 release where we could include new translations.

**Comment by qndel** [comment]
> Why not add the ability to play the PlayStation 1 version of Diablo? 

Can't see why not, do it
